src/org/thoughtcrime/securesms/ConversationListFragment.java

```diff
@@ -161,13 +158,15 @@ public void onResume() {
     super.onResume();
 
     updateReminders();
-    list.getAdapter().notifyDataSetChanged();
+    LoaderManager.getInstance(this).restartLoader(0, null, this);
 
     reloadTimer = new Timer();
     reloadTimer.scheduleAtFixedRate(new TimerTask() {
       @Override
       public void run() {
-        Util.runOnMain(() -> { list.getAdapter().notifyDataSetChanged(); });
+        Util.runOnMain(() ->
+                LoaderManager.getInstance(ConversationListFragment.this)
+                        .restartLoader(0, null, ConversationListFragment.this));
```

Comment:
Without this change, after blocking a contact, the chatlist was not properly refreshed, i.e the chat didn't vanish from the chatlist. I didn't investigate why notifyDataSetChanged didn't do the job, I guess that it only reloads the elements of the list but doesn't check if items were added/removed.

I didn't measure this, but I guess this will make performance a little worse. Although this reloading is only needed when a contact was blocked, so we could think about remembering that we blocked a contact and only restartLoader() if yes.

Reply:
It doesn't get significantly slower (still around 300ms), but after loading the chatlist, but `onBindViewHolder()` is called again for all 10 or so view holders, which takes 250ms. However, this happens in background and there seemed not to be any negative consequences because of this.

So, all in all, should be fine.

Reply:
It's not fine, @adbenitez reported:

"Now when I go back to the chat list I see briefly the chat with the new message badge and an old message as latest message, then it is updated"

Reply:
seems fast now